main.tf

```
@@ -1,13 +1,186 @@
-resource "random_integer" "example" {
+# ----------------------------------------------------------------------------------------------------------------------
+# Enable and configure AWS Config
+# ----------------------------------------------------------------------------------------------------------------------
+module "aws_config_label" {
+  source  = "cloudposse/label/null"
+  version = "0.21.0"
+
+  attributes = ["config"]
+  context    = module.this.context
+}
+resource "aws_config_configuration_recorder" "recorder" {
+  count    = module.this.enabled ? 1 : 0
+  name     = module.aws_config_label.id
+  role_arn = local.create_iam_role ? module.iam_role[0].arn : var.iam_role_arn
+  recording_group {
+    all_supported                 = true
+    include_global_resource_types = var.include_global_resource_types
+  }
+}
+
+resource "aws_config_delivery_channel" "channel" {
+  count          = module.this.enabled ? 1 : 0
+  name           = module.aws_config_label.id
+  s3_bucket_name = module.aws_config_storage[0].bucket_id
+  s3_key_prefix  = "foo"
+  sns_topic_arn  = local.findings_notification_arn
+
+  depends_on = [
+    aws_config_configuration_recorder.recorder,
+    module.aws_config_storage,
+    module.iam_role,
+  ]
+}
+
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Create an S3 Bucket for AWS Config rules to be stored
+# ----------------------------------------------------------------------------------------------------------------------
+module "aws_config_storage" {
   count = module.this.enabled ? 1 : 0
+  #TODO: switch this once published to the registry
+  source = "git::https://github.com/cloudposse/terraform-aws-config-storage.git?ref=feature/initial-implementation"
+
+  force_destroy = var.force_destroy
+
+  context = module.this.context
+}
+
+#-----------------------------------------------------------------------------------------------------------------------
+# Optionally create an SNS topic and subscriptions 
+#-----------------------------------------------------------------------------------------------------------------------
+module "sns_topic" {
+  source  = "cloudposse/sns-topic/aws"
+  version = "0.9.0"
+  count   = local.create_sns_topic ? 1 : 0
+
+  attributes      = ["config"]
+  subscribers     = var.subscribers
+  sqs_dlq_enabled = false
+
+  context = module.this.context
+}
+
+module "aws_config_findings_label" {
+  source  = "cloudposse/label/null"
+  version = "0.21.0"
+
+  attributes = ["config-findings"]
```

possible consideration: the delimiter is configurable in null label and well be adding things like case conversions (e.g.  kebabCase). 

```suggestion
  attributes = ["config", "findings"]
```

---

e3947a3316055a21ace9b41c2f4ad42fce462e1b